lib/metadata/BucketInfo.js

```diff
@@ -68,6 +102,22 @@ export default class BucketInfo {
                 MfaDelete === 'Enabled' ||
                 MfaDelete === 'Disabled');
         }
+        if (websiteConfiguration) {
+            assert.strictEqual(typeof websiteConfiguration, 'object');
+            const { indexDocument, errorDocument, redirectAllRequestsTo,
+                routingRules } = websiteConfiguration;
+            assert(indexDocument === undefined ||
+                typeof indexDocument === 'string');
+            assert(errorDocument === undefined ||
+                typeof errorDocument === 'string');
+            assert(redirectAllRequestsTo === undefined ||
+                typeof redirectAllRequestsTo === 'object');
+            assert(routingRules === undefined ||
+                Array.isArray(routingRules));
+            websiteConfig = websiteConfiguration instanceof
```

Comment:
I don't see the need of fallback, asserting `websiteConfiguration instanceof WebsiteConfiguration` should be enough.